com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -602,149 +642,221 @@ internal bool ApplyTransformToNetworkState(ref NetworkTransformState networkStat
             {
-                interpolatedScale.x = networkState.IsTeleportingNextFrame || !Interpolate ? networkState.Scale.x : m_ScaleXInterpolator.GetInterpolatedValue();
+                interpolatedScale.x = isTeleporting || !Interpolate ? networkState.ScaleX : m_ScaleXInterpolator.GetInterpolatedValue();
             }
 
-            if (SyncScaleY)
+            if (networkState.HasScaleY)
             {
-                interpolatedScale.y = networkState.IsTeleportingNextFrame || !Interpolate ? networkState.Scale.y : m_ScaleYInterpolator.GetInterpolatedValue();
+                interpolatedScale.y = isTeleporting || !Interpolate ? networkState.ScaleY : m_ScaleYInterpolator.GetInterpolatedValue();
             }
 
-            if (SyncScaleZ)
+            if (networkState.HasScaleZ)
             {
-                interpolatedScale.z = networkState.IsTeleportingNextFrame || !Interpolate ? networkState.Scale.z : m_ScaleZInterpolator.GetInterpolatedValue();
+                interpolatedScale.z = isTeleporting || !Interpolate ? networkState.ScaleZ : m_ScaleZInterpolator.GetInterpolatedValue();
             }
 
-            // Position Apply
-            if (SyncPositionX || SyncPositionY || SyncPositionZ)
+            // Apply the new position
+            if (networkState.HasPositionChange)
             {
                 if (InLocalSpace)
                 {
-                    transformToUpdate.localPosition = interpolatedPosition;
+
+                    transform.localPosition = interpolatedPosition;
                 }
                 else
                 {
-                    transformToUpdate.position = interpolatedPosition;
+                    transform.position = interpolatedPosition;
                 }
             }
 
-            // RotAngles Apply
-            if (SyncRotAngleX || SyncRotAngleY || SyncRotAngleZ)
+            // Apply the new rotation
+            if (networkState.HasRotAngleChange)
             {
                 if (InLocalSpace)
                 {
-                    transformToUpdate.localRotation = Quaternion.Euler(interpolatedRotAngles);
+                    transform.localRotation = Quaternion.Euler(interpolatedRotAngles);
                 }
                 else
                 {
-                    transformToUpdate.rotation = Quaternion.Euler(interpolatedRotAngles);
+                    transform.rotation = Quaternion.Euler(interpolatedRotAngles);
                 }
             }
 
-            // Scale Apply
-            if (SyncScaleX || SyncScaleY || SyncScaleZ)
+            // Apply the new scale
+            if (networkState.HasScaleChange)
             {
-                transformToUpdate.localScale = interpolatedScale;
+                transform.localScale = interpolatedScale;
             }
         }
 
-        private void AddInterpolatedState(NetworkTransformState newState, bool reset = false)
+        /// <summary>
+        /// Only non-authoritative instances should invoke this
```

Comment:
Should there be a check in the method to enforce this comment?

Reply:
Really, there is only one method that calls that:
```
        private void OnNetworkStateChanged(NetworkTransformState oldState, NetworkTransformState newState)
        {
            if (!NetworkObject.IsSpawned)
            {
                return;
            }

            if (CanCommitToTransform)
            {
                // we're the authority, we ignore incoming changes
                return;
            }

            if (Interpolate)
            {
                AddInterpolatedState(newState);
            }
        }
```
We really should migrate that code into the OnNetworkStateChanged method.  Was trying to reduce the complexity of the PR (already removed several methods).
It is added to the list.  👍

Reply:
Oh, right, it's private, not protected. Ok, withdrawn.